ionc/ion_binary.c

```diff
@@ -535,6 +546,39 @@ iERR ion_binary_write_float_value( ION_STREAM  *pstream, double value )
     iRETURN;
 }
 
+iERR ion_binary_write_float_value( ION_STREAM  *pstream, float value )
+{
+    iENTER;
+    uint32_t intvalue = 0;
+    BYTE     image[ UINT_32_IMAGE_LENGTH ];
+    BYTE    *pb = &image[UINT_32_IMAGE_LENGTH - 1];
+    int      len;
+
+    ASSERT( UINT_32_IMAGE_LENGTH == 8 );                // we do depend on this here
```

Comment:
This assertion won't pass, will it? Let's just remove it; the next assertion is sufficient.